From 1c2506434e744d8c6a86e42c9d8bae4cde7553f6 Mon Sep 17 00:00:00 2001
From: Mark Asselstine <mark.asselstine@windriver.com>
Date: Thu, 31 May 2018 15:14:26 -0400
Subject: [PATCH] tests: add '--no-validate' when using download template

We are usually running the ptests with core-image-minimal which has no
mechanism to validate the downloads. Validation isn't really of
interest to this test at any rate so simply add '--no-validate' to
avoid failing due to no GPG validation.

Signed-off-by: Mark Asselstine <mark.asselstine@windriver.com>

---
 src/tests/lxc-test-apparmor-mount | 2 +-
 src/tests/lxc-test-autostart      | 2 +-
 src/tests/lxc-test-no-new-privs   | 2 +-
 src/tests/lxc-test-unpriv         | 2 +-
 src/tests/lxc-test-usernic.in     | 2 +-
 5 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/tests/lxc-test-apparmor-mount b/src/tests/lxc-test-apparmor-mount
index d21c948..9e1969b 100755
--- a/src/tests/lxc-test-apparmor-mount
+++ b/src/tests/lxc-test-apparmor-mount
@@ -169,7 +169,7 @@ if [ -f /etc/lsb-release ]; then
     done
 fi
 
-run_cmd lxc-create -t download -n $cname -- -d ubuntu -r $release -a $ARCH
+run_cmd lxc-create -t download -n $cname -- --no-validate -d ubuntu -r $release -a $ARCH
 
 echo "test default confined container"
 run_cmd lxc-start -n $cname -d -lDEBUG -o "$logfile"
diff --git a/src/tests/lxc-test-autostart b/src/tests/lxc-test-autostart
index e5b651b..d15b79b 100755
--- a/src/tests/lxc-test-autostart
+++ b/src/tests/lxc-test-autostart
@@ -55,7 +55,7 @@ if [ -f /etc/lsb-release ]; then
     done
 fi
 
-lxc-create -t download -n $CONTAINER_NAME -B dir -- -d ubuntu -r $release -a $ARCH
+lxc-create -t download -n $CONTAINER_NAME -B dir -- --no-validate -d ubuntu -r $release -a $ARCH
 CONTAINER_PATH=$(dirname $(lxc-info -n $CONTAINER_NAME -c lxc.rootfs.path -H) | sed -e 's/dir://')
 cp $CONTAINER_PATH/config $CONTAINER_PATH/config.bak
 
diff --git a/src/tests/lxc-test-no-new-privs b/src/tests/lxc-test-no-new-privs
index 8642992..e72bdf0 100755
--- a/src/tests/lxc-test-no-new-privs
+++ b/src/tests/lxc-test-no-new-privs
@@ -47,7 +47,7 @@ if type dpkg >/dev/null 2>&1; then
 	ARCH=$(dpkg --print-architecture)
 fi
 
-lxc-create -t download -n c1 -- -d ubuntu -r xenial -a $ARCH
+lxc-create -t download -n c1 -- --no-validate -d ubuntu -r xenial -a $ARCH
 echo "lxc.no_new_privs = 1" >> /var/lib/lxc/c1/config
 
 lxc-start -n c1
diff --git a/src/tests/lxc-test-unpriv b/src/tests/lxc-test-unpriv
index 16ff12d..0958d48 100755
--- a/src/tests/lxc-test-unpriv
+++ b/src/tests/lxc-test-unpriv
@@ -173,7 +173,7 @@ run_cmd mkdir -p $HDIR/.cache/lxc
     cp -R /var/cache/lxc/download $HDIR/.cache/lxc && \
     chown -R $TUSER: $HDIR/.cache/lxc
 
-run_cmd lxc-create -t download -n c1 -- -d ubuntu -r $release -a $ARCH
+run_cmd lxc-create -t download -n c1 -- --no-validate -d ubuntu -r $release -a $ARCH
 
 # Make sure we can start it - twice
 
diff --git a/src/tests/lxc-test-usernic.in b/src/tests/lxc-test-usernic.in
index 3e35008..f489286 100755
--- a/src/tests/lxc-test-usernic.in
+++ b/src/tests/lxc-test-usernic.in
@@ -146,7 +146,7 @@ if [ -f /etc/lsb-release ]; then
 fi
 
 # Create three containers
-run_cmd "lxc-create -t download -n b1 -- -d ubuntu -r $release -a $ARCH"
+run_cmd "lxc-create -t download -n b1 -- --no-validate -d ubuntu -r $release -a $ARCH"
 run_cmd "lxc-start -n b1 -d"
 p1=$(run_cmd "lxc-info -n b1 -p -H")
 
